tests/unit/test_utils.py

```diff
@@ -592,3 +594,74 @@ def test_check_requires(self, metadata, should_raise):
                 check_dist_requires_python(fake_dist)
         else:
             check_dist_requires_python(fake_dist)
+
+
+class TestValidatePath(object):
+    def setup(self):
+        self.tempdir = tempfile.mkdtemp()
+        self.old_mask = os.umask(0o022)
+        self.dir = DATA_DIR.join('util').join('bar')
+        self.file = self.dir.join('foo')
+        self.symlinksDir = Path(self.tempdir)
+
+    def teardown(self):
+        os.umask(self.old_mask)
+        shutil.rmtree(self.tempdir, ignore_errors=True)
+
+    def testRegularFile(self):
+        foo_file = self.file.join('foo')
+        validate_path(foo_file)
+
+    @pytest.mark.skipif(
+        not hasattr(os, 'symlink'),
+        reason="requires os.symlink",
+    )
+    def testRegularSymlinkToFile(self):
+        foo_file = self.file
+        symfoo_link = self.symlinksDir.join('symfoo')
+        try:
+            os.symlink(foo_file, symfoo_link)
+            validate_path(symfoo_link)
+        except OSError:
+            return
```

Comment:
Skip the test or make it fail if you come inside the except; maybe don't actually catch it?

Same for later cases as well.